### PR TITLE
Style Manage List Page and Add Toast Alerts

### LINKED ISSUE #50: 25. Style Manage List Page and Add Toast Alerts
## Description
Style the manage list page such that the purchase urgency options both look professional and are accessible. Additionally, replace the JavaScript alerts with toasts.

## Acceptance Criteria
- [x] Purchase urgency options are fully accessible.
- [x] Appropriate toast alerts display when successfully creating a list, adding an item, and sharing a list.
- [x] Appropriate toast alerts display when the above actions fail.

### PR DESCRIPTION
**Pull request by jaguilar89**
## Description
This code adds stylish and accessible radio buttons to the AddList component using Radix's RadioGroup component. Additionally, I thought this was good opportunity to get rid of the old alerts so toast alerts have been added to replace them.

### Notes
- Right now all toasts have a light theme by default, this can be configured to change conditionally once the dark mode branch is merged.
- The toasts appear to be missing their icons, at least on my computer. Let me know if the same is true for you guys. For example the success and error alerts should look like the following:
image
image

## Related Issue

Closes #50 

## Acceptance Criteria
- [x] Purchase urgency options are fully accessible.
- [x] Appropriate toast alerts display when successfully creating a list, adding an item, and sharing a list.
- [x] Appropriate toast alerts display when the above actions fail.

## Type of Changes

Enhancement, accessibility

### Before
Screenshot 2024-04-02 at 9 07 17 PM
Screenshot 2024-04-02 at 9 45 54 PM
Screenshot 2024-04-02 at 9 45 34 PM

### After
Screenshot 2024-04-02 at 9 12 19 PM

https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/78065152/7fdeb6fd-1990-4a5d-818a-6a80f5a78089

## Testing Steps / QA Criteria
`git pull`
`git checkout ja-style-manage-list`
`npm ci`
`npm start`

- Navigate to the Manage List page, test the radio buttons for full keyboard accessibility.
- Perform various actions and test toast alert visibility (create a list, add new item, etc)
- Test error alert visibility by attempting to add a nameless item, share a list with an invalid email, etc.

### CODE REVIEW

**Review by trushmi — COMMENTED** [review]
Hi Jose! Thank you for addressing this issue and implementing Toast Alerts on our website to enhance the user experience! Regarding your notes about icons, they are visible and look great! Speaking of toast alerts, they function properly on my machine in certain cases.

**Manage List page:**

Upon adding a new item to the list, I receive a message indicating its successful addition. However, if I immediately press the 'Submit' button thereafter, I encounter a message stating the item is a duplicate, even though my input field is empty after submission. Additionally, when adding an email, an old alert message persists when attempting to add an incorrect email.

**Home page:**

When attempting to add an empty string instead of a list name, I only see a message: "Please fill out this field," triggered by 'required' as a property for the input instead of a toast alert.

Upon creating a list, I do receive a toast alert as expected. However, I am redirected to the list page faster than the alert appears for the first time. Perhaps we could consider implementing a loader to provide a smoother user experience?

**List page:**
Upon deleting an item from the list, an old alert message appears.

To ensure the best user experience, could we update the files and incorporate Toast Alerts in all instances where we display messages to the user across all pages?

**Radio buttons**
Regarding radio buttons, I find them well-styled! Nice work!!

**Review by rachelspencer — COMMENTED** [review]
Hey Jose!

Nice job working with the radio buttons! The toasts look really cool! There are a  couple of things to tweak. 

Create a list toast
Same as Iryna's comment on the required for an empty submission, can we make this an alert? Nitpick: For an empty string toast if you have time, could the error message be updated to "Error occurred. List not created, enter a valid list name"? 

Deleting an item toast
Old alerts still show. We had a lot of alerts, so this one probably slipped through he cracks!

Adding an item toast
If I add a new item that isn't in the current list, it adds successfully (the toast looks great). If I try to submit without putting anything in the input field (not even a space) I get a toast saying that the item is already in the list (same as Ityna). Maybe there is some state that is storing the last item input value and its not clearing after submission?

Share email toast
If an email doesn't exist in the users, the old alter is still appearing. If the email is invalid I am getting the old alert that says the user isn't in the system.

Great job working with toasts for the first time Jose! There was a lot to do in this issue as we had a lot of alerts, so well done!

**Comment by jaguilar89** [comment]
> Hey Jose!
> 
> Nice job working with the radio buttons! The toasts look really cool! There are a couple of things to tweak.
> 
> Create a list toast Same as Iryna's comment on the required for an empty submission, can we make this an alert? Nitpick: For an empty string toast if you have time, could the error message be updated to "Error occurred. List not created, enter a valid list name"?
> 
> Deleting an item toast Old alerts still show. We had a lot of alerts, so this one probably slipped through he cracks!
> 
> Adding an item toast If I add a new item that isn't in the current list, it adds successfully (the toast looks great). If I try to submit without putting anything in the input field (not even a space) I get a toast saying that the item is already in the list (same as Ityna). Maybe there is some state that is storing the last item input value and its not clearing after submission?
> 
> Share email toast If an email doesn't exist in the users, the old alter is still appearing. If the email is invalid I am getting the old alert that says the user isn't in the system.
> 
> Great job working with toasts for the first time Jose! There was a lot to do in this issue as we had a lot of alerts, so well done!

@rachelspencer Thanks for the feedback! It seems I missed some alerts. I'll take another look at this asap.

**Review by sillytsundere — COMMENTED** [review]
The toasts look great and this really adds to the look and feel of the app! Thanks for doing all this work! The radio buttons work great, can navigate with my keyboard, but I noticed the same issues that Iryna and Rachel did. 

I also noted that when sharing a list, if it is shared successfully, the old alert is still coming up.

**Comment by jaguilar89** [comment]
@rachelspencer 
 > Create a list toast Same as Iryna's comment on the required for an empty submission, can we make this an alert? Nitpick: For an empty string toast if you have time, could the error message be updated to "Error occurred. List not created, enter a valid list name"?
 
 All set!
> Adding an item toast If I add a new item that isn't in the current list, it adds successfully (the toast looks great). If I try to submit without putting anything in the input field (not even a space) I get a toast saying that the item is already in the list (same as Ityna). Maybe there is some state that is storing the last item input value and its not clearing after submission?

Yeah not sure what's up with that tbh.
> 

Other than that, I replaced the email validator text to a red badge, that way it pops out more. Let me know if it looks ok.

@trushmi @sillytsundere I replaced the alert for confirming an item delete with a modal. I also thought the delete buttons would look better as Radix buttons.. do another pull and let me know what you guys think of it.

**Comment by rachelspencer** [comment]
Hey @jaguilar89 

I think the bug for when a user submits an empty string and it comes up with the duplicate toast should be addressed. Did you make any headway?

With regards to the delete button, I have styled them on my issue when I did the List page styling. @trushmi has attached the classname I created in the global css file to her delete buttons on the Home Page. So when we merge my styling will carry over to the Home page. Maybe we can see how this looks after the merged and see if we want to replace them with the Radix button? This is the current styling:

Screenshot 2024-04-03 at 7 00 08 PM

That being said I would say to leave the delete button as is. 

I am all for the delete modal, it looks great! I like how the toast appears after the user confirms the deletion. 

I like the red text under the email input. Only suggestion here is to make the text black, similar to how I overrode the badge text settings to black as it makes them more accessible. 

Nice job!

**Comment by jaguilar89** [comment]
@rachelspencer 
> I think the bug for when a user submits an empty string and it comes up with the duplicate toast should be addressed. Did you make any headway?

Fixed! Forgot that useRef doesn't trigger re-renders so the itemName variable wasn't resetting.

> That being said I would say to leave the delete button as is.

Got it. I've set it back to the previous button.

> I like the red text under the email input. Only suggestion here is to make the text black, similar to how I overrode the badge text settings to black as it makes them more accessible.

Done!

**Review by sillytsundere — APPROVED** [review]

**Comment by rachelspencer** [comment]
Approved!

**Review by rachelspencer — APPROVED** [review]

**Review by trushmi — APPROVED** [review]
Looks good to me!

File: src/components/AddItem.jsx [hunk 1]
```diff
@@ -1,47 +1,45 @@
...
 
 			await addItem(listPath, {
 				itemName,
 				daysUntilNextPurchase,
 			});
-
-			alert('Item saved successfully');
+			toast.success(`${itemName} added to the list! 👍`);
 			formRef.current.reset();
+			setItemName('');
```
Comment (polly89) [inline src/components/AddItem.jsx:40]: Good catch here.

File: src/components/AddItem.jsx [hunk 2]
```diff
@@ -67,44 +66,27 @@ export function AddItem({ listPath, data }) {
...
+					<RadioGroup.Item id="soon" onClick={handleDays} value={7}>
+						Soon (7 Days)
+					</RadioGroup.Item>
+					<RadioGroup.Item id="kind-of-soon" onClick={handleDays} value={14}>
+						Kind of Soon (14 Days)
+					</RadioGroup.Item>
+					<RadioGroup.Item id="not-soon" onClick={handleDays} value={30}>
+						Not Soon (30 Days)
+					</RadioGroup.Item>
+				</RadioGroup.Root>
```
Comment (polly89) [inline src/components/AddItem.jsx:79]: These worked great!

File: src/index.jsx [hunk 3]
```diff
@@ -23,6 +25,21 @@ root.render(
 	<StrictMode>
 		<DarkModeProvider>
 			<AppWithTheme />
+			<ToastContainer
+				limit={2}
+				position="top-right"
+				icon={true}
+				autoClose={3000}
```
Comment (polly89) [inline src/index.jsx:32]: I would increase this to 4000, I had a hard time reading some of the longer messages in the time alloted.

**Review by polly89 — APPROVED** [review]
Great job! I left one non-blocking comment.

File: src/index.jsx (hunk 3 above)
Comment (rachelspencer) [reply src/index.jsx:32]: We will apply this change! Good thought!

**Pull request merged**